### PR TITLE
add lavaland real

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
adds lavaland for shaft miners to have a permanent source of ore.
when mapped or an admeme spawns it, the `Mining, Filled` suffix shuttle airlock will spawn a mining shuttle, you can use it to FTL. LO gets a spare console board (which replaces the dud board that was obsoleted years ago)

lavaland has no dungeon but unlimited ore, the same mined from expeditions

## Why / Balance
hopefully salv will be less shit when they have roundstart exped-tier mining, though eventually you will run out of ore near the base and either have to go further and further out or go on expeds to get huge bursts of ore again

in the future it can track inserting ore into an ore silo by giving your id mining points, then you can buy gamer loot with it after you do your only job. same for megafauna and dungeons i think

## Technical details
added PlanetPrototype that stores everything for a given planet
new DockingConsoleComponent, lets you go from point A to point B and thats it. can be reused for labor camp and admeme things, whitelisting each location and shuttle separately (lavaland uses MiningShuttleComponent for example)
ftl state doesnt update properly and you have to reopen console a lot because of how ftl is coupled to ShuttleConsoleComponent but ill overhaul it later :trollface:

## Mapping
`external airlock [Mining, Filled]` spawns the 5x7 mining shuttle and automatically docks to it. this should be added to every map.
`mining shuttle console` can remotely fly the ship back to where you need it. this should also be added to every map.
the other one that doesnt have filled in its suffix will be a priority dock too but won't spawn a shuttle. this is what is used on the mining base

if you want to change the planet add a new prototype to `Resources/Prototypes/DeltaV/planets.yml` then add this to the map prototype's components:
```yml
- type: StationPlanetSpawner
  planet: EpicPlanet
```

likewise to use a different base, make a base and **use /savegrid to save it** then add it to the spawner component:
```yml
- type: StationPlanetSpawner
  gridPath: /Maps/Nonstations/DeltaV/epic-base.yml
```

## Media

**Welcome to Lavaland**

notable features:
- comms server for common and logistics radio
- airlock has a vacuum setup to pump out hot, and in the future potentially toxic, air when entering the base from outside
- basic medbay for when you clip the very fair lava hitbox and get instacrit
- spare mining supplies and suits for when LO sends down tiders with no gear to mine
- nourishment for borgs and crew alike
- a fax machine to send "JOHN SMITH LING KILLING AT MINING BASE HELP!!!" to the bridge and for nobody to read it

the mining shuttle can't be expanded so nobody turns it into a mini station, its purely for transit to and from the base

poor tider with no suit is slowly suffocating + uncomfortably hot
(now 80C and 120kPa)

"beautiful" console ui

not quite lavaland...

## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
:trollface:

**Changelog**
:cl:
- add: Added Lavaland, coming soon to a station near you!
- remove: Removed the random reclaimer wrecks.
- fix: Fixed not being able to build catwalks over lava.

### CODE REVIEW

**Comment by MilonPL** [comment]
holy fuck you were not kidding

**Comment by Campbelle1000** [comment]
Holy shit its real, lavaland syndicate base pr coming soon after this!

**Comment by therealDLondon** [comment]
He did it. That crazy son of a bitch did it.

**Comment by Verbalase** [comment]
new favorite server

**Comment by deltanedas** [comment]
ok goober

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleBoundUserInterface.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if (state is DockingConsoleState cast)
+            _window?.UpdateState(cast);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        base.Dispose(disposing);
+
+        if (disposing)
+            _window?.Dispose();
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleBoundUserInterface.cs]: don't dispose, call Orphan

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+public sealed partial class DockingConsoleWindow : FancyWindow
+{
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    private readonly AccessReaderSystem _access;
+
+    public event Action<int>? OnFTL;
+
+    private EntityUid _owner;
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs]: readonly

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    private int? _selected;
+    private FTLState _state;
+    private StartEndTime _ftlTime;
+    private StyleBoxFlat _ftlStyle;
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs]: readonly + sort these

File: Content.Server/DeltaV/Planet/PlanetSystem.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+public sealed class PlanetSystem : EntitySystem
+{
+    [Dependency] private readonly AtmosphereSystem _atmos = default!;
+    [Dependency] private readonly BiomeSystem _biome = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly MapSystem _map = default!;
+    [Dependency] private readonly MapLoaderSystem _mapLoader = default!;
+    [Dependency] private readonly MetaDataSystem _meta = default!;
+
+    private List<(Vector2i, Tile)> _setTiles = new();
```
Comment (MilonPL) [inline Content.Server/DeltaV/Planet/PlanetSystem.cs]: readonly

File: Content.Server/DeltaV/Shuttles/Systems/DockingConsoleSystem.cs [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+using Content.Shared.DeltaV.Shuttles;
+using Content.Shared.DeltaV.Shuttles.Components;
+using Content.Shared.DeltaV.Shuttles.Systems;
+using Content.Shared.Shuttles.Components;
+using Content.Shared.Shuttles.Systems;
+using Content.Shared.Timing;
+using Content.Shared.Whitelist;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using System.Linq;
```
Comment (MilonPL) [inline Content.Server/DeltaV/Shuttles/Systems/DockingConsoleSystem.cs]: remove

File: Content.Server/DeltaV/Shuttles/Systems/DockingConsoleSystem.cs [hunk 6]
```diff
@@ -0,0 +1,166 @@
+using Content.Server.Shuttles.Components;
+using Content.Server.Shuttles.Events;
+using Content.Server.Shuttles.Systems;
+using Content.Server.Station.Components;
```
Comment (MilonPL) [inline Content.Server/DeltaV/Shuttles/Systems/DockingConsoleSystem.cs]: remove

File: Content.Server/DeltaV/Shuttles/Systems/DockingShuttleSystem.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+                Name = Name(mapUid),
+                Map = map.MapId
+            });
+        }
+    }
+
+    private void OnFTLStarted(Entity<DockingShuttleComponent> ent, ref FTLStartedEvent args)
+    {
+        _console.UpdateConsolesUsing(ent);
+    }
```
Comment (MilonPL) [inline Content.Server/DeltaV/Shuttles/Systems/DockingShuttleSystem.cs:51]: this isn't ever being used

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+using Content.Shared.DeltaV.Shuttles;
+using Content.Shared.DeltaV.Shuttles.Components;
+using Content.Shared.Shuttles.Components;
+using Content.Shared.Shuttles.Systems;
+using Content.Shared.Timing;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs]: remove

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Map;
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs]: remove

File: Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
+using Content.Client.UserInterface.Controls;
+using Content.Shared.Access.Systems;
+using Content.Shared.DeltaV.Shuttles;
+using Content.Shared.DeltaV.Shuttles.Components;
+using Content.Shared.Shuttles.Components;
```
Comment (MilonPL) [inline Content.Client/DeltaV/Shuttles/UI/DockingConsoleWindow.xaml.cs]: remove

File: Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
+using Content.Shared.DeltaV.Shuttles.Systems;
+using Content.Shared.Tag;
+using Content.Shared.Whitelist;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs]: remove

File: Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    /// If this is null a shuttle was not found and this console does nothing.
+    /// </summary>
+    [DataField]
+    public EntityUid? Shuttle;
+
+    /// <summary>
+    /// Whether <see cref="Shuttle"/> is set on the server or not.
+    /// Client can't use Shuttle outside of PVS range so that isn't networked.
+    /// </summary>
+    [DataField, AutoNetworkedField]
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs:46]: why does this need to be networked if it's never being used on the client

File: Content.Shared/DeltaV/Shuttles/DockingConsoleUI.cs [hunk 13]
```diff
@@ -0,0 +1,27 @@
+using Content.Shared.DeltaV.Shuttles.Components;
+using Content.Shared.Shuttles.Systems;
+using Content.Shared.Timing;
+using Robust.Shared.Map;
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Shuttles/DockingConsoleUI.cs]: remove

File: Resources/Prototypes/DeltaV/Entities/Structures/Machines/computers.yml [hunk 14]
```diff
@@ -29,3 +29,51 @@
...
+    - map: ["computerLayerBody"]
+      state: computer
+    - map: ["computerLayerKeyboard"]
+      state: generic_keyboard
+    - map: ["computerLayerScreen"]
+      state: shuttle
+    - map: ["computerLayerKeys"]
+      state: generic_keys
+    - map: [ "enum.WiresVisualLayers.MaintenancePanel" ]
+      state: generic_panel_open
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Structures/Machines/computers.yml]: spaces in inline lists everywhere, not just on the last one
`[ "computerLayerBody" ]`

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Server/DeltaV/Shuttles/Systems/DockingShuttleSystem.cs (hunk 7 above)
Comment (deltanedas) [reply Content.Server/DeltaV/Shuttles/Systems/DockingShuttleSystem.cs:51]: forgor

File: Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs (hunk 12 above)
Comment (deltanedas) [reply Content.Shared/DeltaV/Shuttles/Components/DockingConsoleComponent.cs:46]: meant to have a thing that displays an error when theres no shuttle :trollface:

File: Resources/Prototypes/DeltaV/Entities/Structures/Machines/computers.yml (hunk 14 above)
Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Structures/Machines/computers.yml]: john copy paste

**Comment by Avalon-Proto** [comment]
This is gonna be so peak. Question with the "Glacier Surface" though. Will this be 
A) The station is now on land. So you can just freely walk around
Or
B) Glacier's version of lavaland

**Comment by deltanedas** [comment]
its B

**Comment by CRISOOOOO** [comment]
Lavaland ruins will be real too?

**Comment by Bonktrauma** [comment]
> Lavaland ruins will be real too?

Wyci probably

**Review by MilonPL — DISMISSED** [review]
looks good code-wise, direction feedback pending

**Comment by Lyndomen** [comment]
Direction review: Mostly good, some thoughts:

image
Why is lavaland so tolerable? While the oxy is low, this is perfectly survivable by your average cargo tech with a mask. Could you try making the planet a bit hotter, and the base more reliant on refrigeration? 

ghost view is fucked up, maybe thats just a feature i dont really care
image

Shuttle console is not access locked, nor is the shuttle door itself. Ideally they would have externals at the least. The base also is entirely all access. I dont really want the entire station flooding through and jumping in lava every round, it should be a real threat.

**Comment by Bonktrauma** [comment]
> Direction review: Mostly good, some thoughts:
> 
> image
> Why is lavaland so tolerable? While the oxy is low, this is perfectly survivable by your average cargo tech with a mask. Could you try making the planet a bit hotter, and the base more reliant on refrigeration? 
> 
> ghost view is fucked up, maybe thats just a feature i dont really care
> image
> 
> Shuttle console is not access locked, nor is the shuttle door itself. Ideally they would have externals at the least. The base also is entirely all access. I dont really want the entire station flooding through and jumping in lava every round, it should be a real threat. 

How many urists did you throw in the lava?

**Comment by deltanedas** [comment]
> ghost view is fucked up

thats just how generation works, it ignores ghosts so you cant ddos by flying around

**Review by Lyndomen — APPROVED** [review]

**Review by MilonPL — APPROVED** [review]

**Comment by Carolyn3114** [comment]
wait, so, why not remove expeditions and salvage magnet?

**Comment by deltanedas** [comment]
magnet debris is still great and rocks can be used for a quick burst of materials
eventually all the ore near the base will get mined so you have to chose either to go further and further, leaving you in danger if you step in lava far away, or to do exped for a fresh supply of ore

**Comment by CrazyPhantom779** [comment]
Hello! I'm a Developer on the Starlight fork. I was wondering if you would be willing to re-license this PR to MIT?

**Pull request merged**